### PR TITLE
Creating package

### PR DESCRIPTION
**Pull request by or-yam**
@davidmeirlevy Please take a look and let me know what you think. 
I didn't completely understand how to auto-generate configuration from environment variables.
Once we done with the review I will publish it to npm.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+const app = require('express');
```
Comment (davidmeirlevy) [inline index.js:1]: no.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  proxyReq.setHeader('user', req.user || '');
+}
+
+/**
+ * This function is a temporary thingy because I'm too lazy right now
+ * to insert an nginx router to the whole application.
+ * when the nginx router will be added - this file can be removed.
+ * @param app: Express.Application
+ */
+module.exports = function apiProxy(config) {
```
Comment (davidmeirlevy) [inline index.js:21]: this function should get the app as a first parameter, and the config is second.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  const { authService, contentService, adminPanel, assetsService, tenant, draftsService, applicationUrl } = config;
```
Comment (davidmeirlevy) [inline index.js:22]: this package should also build its own config.
you should also take those objects from the original config file at the `blog-front` and put them inside this package (use a config.js file).

File: index.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        const setCookie = response.headers.raw()['set-cookie'];
+        if (setCookie) {
+          res.set('set-cookie', setCookie);
+        }
+        if (response.status === 200) {
+          return response.text();
+        }
+      })
+      .then(user => {
+        req.user = user;
```
Comment (davidmeirlevy) [inline index.js:75]: I found out a better way of doing this line :)

you can put this user on the headers directly, like: `req.headers.user = user` and then you can also remove the `onProxyReq ` function, and its usage.

File: index.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  const APP_HOSTNAME = applicationUrl ? getPureHostname(applicationUrl) : null;
+
+  function getProxy(target) {
+    return proxy({
+      target,
+      changeOrigin: true,
+      headers: {
+        tenant
+      },
+      onProxyReq
```
Comment (davidmeirlevy) [inline index.js:32]: read my comment regarding this function :)

File: package.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (davidmeirlevy) [inline package.json:1]: look at this package.json for a good reference to a package:
https://github.com/greenpress/api-kit/blob/master/package.json

- no need for scripts
- no need for "nodemon"
- no need "express" as a dependency (but as a peerDependency).
- the license should be MIT.
- missing "main" property here.
- also take the license, contributing, .editorconfig, .gitignore, and readme file from the api-kit package.

**Review by davidmeirlevy — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (or-yam) [reply index.js:22]: Do you mean something like [this](https://github.com/greenpress/blog-front/blob/master/config/index.js) file ?

Comment (davidmeirlevy) [reply index.js:17]: Yes

File: package.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  "author": "David Meir-Levy <davidmeirlevy@gmail.com>",
+  "homepage": "https://github.com/greenpress/api-proxy-middleware#readme",
+  "private": false,
+  "main": "index.js",
+  "dependencies": {
+    "http-proxy-middleware": "^1.3.1",
+    "node-fetch": "^2.6.1"
+  },
+  "peerDependencies": {
+    "express": "^4.17.1"
```
Comment (davidmeirlevy) [inline package.json:14]: At peer you should use ">=x.y.z"

**Review by davidmeirlevy — CHANGES_REQUESTED** [review]
Last minor changes..

**Comment by or-yam** [comment]
@davidmeirlevy Updated with the requested changes.

File: config.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
+module.exports = {
+  port: process.env.PORT || 3000,
+  tenant: process.env.BASIC_TENANT || "0",
+  applicationUrl: process.env.APPLICATION_URL || "http://localhost:3000",
+  alternativeHmr: !!process.env.ALTERNATIVE_HMR,
```
Comment (davidmeirlevy) [inline config.js:5]: This variable isn't needed here.

File: config.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
+module.exports = {
+  port: process.env.PORT || 3000,
```
Comment (davidmeirlevy) [inline config.js:2]: This variable isn't needed here.

File: index.js [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+/**
+ * This function is a temporary thingy because I'm too lazy right now
+ * to insert an nginx router to the whole application.
+ * when the nginx router will be added - this file can be removed.
+ * @param app: Express.Application
+ */
+module.exports = function apiProxy(app, config) {
+  const services = Object.keys({ ...configFile }).map((service) => config[service] || configFile[service]);
```
Comment (davidmeirlevy) [inline index.js:18]: There are better ways to extend an object in JS.

**Review by davidmeirlevy — CHANGES_REQUESTED** [review]

File: index.js (hunk 10 above)
Comment (or-yam) [reply index.js]: Fixed.

**Review by davidmeirlevy — APPROVED** [review]

**Pull request merged**